src/kubernetes_multus.py

```diff
@@ -0,0 +1,276 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm Library used to leverage the Multus Kubernetes CNI in charms.
+
+- On charm installation, it will:
+  - Create the requested network attachment definitions
+  - Patch the statefultset with the necessary annotations for the container to have interfaces
+    that use those new network attachments.
+- On charm removal, it will:
+  - Delete the created network attachment definitions
+
+## Usage
+
+```python
+
+from kubernetes_multus import (
+    KubernetesMultusCharmLib,
+    NetworkAttachmentDefinition,
+    NetworkAnnotation
+)
+
+class YourCharm(CharmBase):
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self._kubernetes_multus = KubernetesMultusCharmLib(
+            charm=self,
+            network_attachment_definitions=[
+                NetworkAttachmentDefinition(name="access-net"),
+                NetworkAttachmentDefinition(name="core-net"),
+            ],
+            network_annotations=[
+                NetworkAnnotation(name="access-net", interface="eth0"),
+                NetworkAnnotation(name="core-net", interface="eth1"),
+            ]
+        )
+```
+"""
+
+import json
+import logging
+from dataclasses import asdict, dataclass, field
+from typing import List, Optional, Union
+
+import httpx
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.generic_resource import create_namespaced_resource
+from lightkube.models.meta_v1 import ObjectMeta
+from lightkube.resources.apps_v1 import StatefulSet
+from lightkube.types import PatchType
+from ops.charm import CharmBase, InstallEvent, RemoveEvent, UpgradeCharmEvent
+from ops.framework import Object
+
+logger = logging.getLogger(__name__)
+
+NetworkAttachmentDefinitionResource = create_namespaced_resource(
+    group="k8s.cni.cncf.io",
+    version="v1",
+    kind="NetworkAttachmentDefinitionResource",
+    plural="network-attachment-definitions",
+)
```

Comment:
Am I right in assuming this will create K8s resources on import? I would avoid that and probably move it in `__init__`.

Reply:
No this method only creates a class, not an actual kubernetes resource